### PR TITLE
Chat Service

### PR DESCRIPTION
**Pull request by ben-toogood**

### CODE REVIEW

**Comment by asim** [comment]
Can you add a README like the blog that explains what this is, how to use it, etc

File: chat/README.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+# Chat Service
+
+The chat service is an example Micro service which leverages bidirectional streaming, the store and events to build a chat backend. There is both a server and client which can be run together to demonstrate the application (see client/main.go for more instructions on running the service).
+
+The service is documented inline and is designed to act as a reference for the events package.
```
Comment (asim) [inline chat/README.md:5]: Any example usage? Can you fill it out. Like Janos is doing with the blog https://github.com/micro/services/tree/master/blog

Comment (ben-toogood) [reply chat/README.md:5]: I documented pragmatic usage in the client/main.go. It's not something you can call easily via the CLI / API. We should implement bidirectional streaming in the CLI and possibly even add streaming to the API?

Comment (asim) [reply chat/README.md:5]: API supports streaming via web sockets, CLI has a `micro stream` command but I think bidrectional streaming will require more work. I think it highlights this service needs unary endpoints also and then at the very least those documented.

**Pull request merged**